### DISCUSSION TITLE
Watching for PRs

### DISCUSSION THREAD

**Discussion by Disinterpreter** [multitheftauto/mtasa-blue#1929] [General]
Right now we have 83 PRs for the different time from different people. And I think it's not a good for us.

I suggest to discuss the best ideas about How to improve our management system of PRs.

Reasons of this step is: Making a point of the best management of PRs. (looking for stale PRs, alerting when PR is ready to merge etc..)

What can I suggest:
1. Pay attention on old PRs. I mean, many PRs was created for two years ago and it invalid for now. I suppose we need to close them.
2. Implement some bots which would attach labels of doings authors (like when author write I NEED HELP bot attach a label "need help", when author write "I'm done" the bot attach a label "done"), or the bots which will watch and alert about old PRs which ready to merge for closing or merged it. (ex: https://github.com/googleapis/repo-automation-bots)

Please, if you had experience with management of huge projects write here and put your ideas.

Thank you!

**Comment by sbx320** [comment]
The older PRs usually still have a bit of valuable information to them, so I'd be a bit hesitant to just close them for inactivity, since someone else might want to pick them up again. Many just need a quick rebase to master to become mergable again. A couple of them are also blocked on network compatibility issues, which would just be rebased on master if we ever get to 1.6. Ultimately we'd just need more people to review PRs. This doesn't even need to be someone with commit/merge access. If you want to review a PR, go ahead. The cleaner the PR is before someone with merge access looks at them, the better. 

Personally my main issue for tracking PRs is that I don't really monitor Github notifications (too much noise) so I commonly miss updates (e.g. requested changes being applied). So the easiest way to get a timely review is to poke me on Discord. A bot might help here as well to indicate a "PR done" state.

**Comment by patrikjuvonen** [comment]
Also keep in mind many PRs have been submitted as drafts and are still counted into the total PRs. If people keep posting drafts for comments but never intend to finish them, we will end up with lots of PRs.

What also matters is yes we might have 80+ open PRs but we have 720+ closed PRs. The ratio is good, we aren't doomed. Just that a lot of the existing PRs are waiting for something or someone.

I agree though we may need some system to automate pings maybe, to keep people active so they don't stay for years.

![Untitled](https://user-images.githubusercontent.com/22572159/104846178-e8b51700-58e1-11eb-846c-e2fb1cb00d19.png)
Merge + close is positive which is great!

**Comment by Disinterpreter** [comment]
Should we use "priority" labels for some bugs and features? It really can help new people which don't know how they can help us.

**Comment by Pirulax** [comment]
I think the main blocker (especially with PR that affect some core component / lots of code) is the fear of breaking something, in which unit tests would help a lot. See discussion [here](https://github.com/multitheftauto/mtasa-blue/discussions/2190).
